tests/manager/celery/tasks/test_opds_for_distributors.py

```diff
@@ -0,0 +1,315 @@
+from tests.fixtures.http import MockHttpClientFixture
+from tests.fixtures.redis import RedisFixture
+
+
+class OPDSForDistributorsImportFixture:
+    def __init__(
+        self,
+        db: DatabaseTransactionFixture,
+        http_client: MockHttpClientFixture,
+        apply_fixture: ApplyTaskFixture,
+        files: OPDSForDistributorsFilesFixture,
+    ):
+        self.db = db
+        self.collection = self.db.collection(
+            protocol=OPDSForDistributorsAPI,
+            settings=OPDSForDistributorsSettings(
+                username="a",
+                password="b",
+                data_source="data_source",
+                external_account_id="http://opds",
+            ),
+        )
+        self.client = http_client
+        self.apply = apply_fixture
+        self.files = files
+
+    def run_import_task(self, collection: Collection | None = None) -> None:
+        collection = collection if collection is not None else self.collection
+        opds_for_distributors.import_collection.delay(collection.id).wait()
+        self.apply.process_apply_queue()
+
+    def authentication_document(self) -> str:
+        """Return the authentication document URL for the collection."""
+        return AuthenticationDocument(
+            id="http://test-authentication-document",
+            title="Test Authentication Document",
+            authentication=[
+                Authentication(
+                    type="http://opds-spec.org/auth/oauth/client_credentials",
+                    links=[Link(href=self.db.fresh_url(), rel="authenticate")],
+                )
+            ],
+        ).model_dump_json()
+
+    def token(self) -> str:
+        return OAuthTokenResponse(
+            access_token="token",
+            token_type="Bearer",
+            expires_in=3600,
+        ).model_dump_json()
+
+    def queue_up_auth_responses(self) -> None:
+        """Queue up the authentication document and token responses."""
+        self.client.queue_response(
+            200,
+            content=self.files.sample_data("biblioboard_mini_feed.opds"),
+        )
+        self.client.queue_response(
+            200,
+            content=self.authentication_document(),
+            headers={"Content-Type": AuthenticationDocument.content_type()},
+        )
+        self.client.queue_response(
+            200,
+            content=self.token(),
+            headers={"Content-Type": "application/json"},
+        )
+
+
+@pytest.fixture
+def opds_for_distributors_import_fixture(
+    db: DatabaseTransactionFixture,
+    http_client: MockHttpClientFixture,
+    apply_task_fixture: ApplyTaskFixture,
+    opds_dist_files_fixture: OPDSForDistributorsFilesFixture,
+    celery_fixture: CeleryFixture,
+    redis_fixture: RedisFixture,
+) -> OPDSForDistributorsImportFixture:
+    return OPDSForDistributorsImportFixture(
+        db, http_client, apply_task_fixture, opds_dist_files_fixture
+    )
+
+
+class TestImportCollection:
```

Comment:
Most of the tests here were moved over from the `TestOPDSForDistributorsImporter` class.